src/transform/function_value_body.cc

```diff
@@ -2,18 +2,19 @@
 
 #include "ast/function_declaration.h"
 #include "ast/return_statement.h"
+#include "ast/statement.h"
+#include "ast/block.h"
 
 namespace transform {
 void FunctionValueBodyTransformer::visit(ast::FunctionDeclaration* node) {
   using ValueBody = ast::FunctionDeclaration::ValueBody;
-  using StatementsBody = ast::FunctionDeclaration::StatementsBody;
   if (node->body().is<ValueBody>()) {
     std::unique_ptr<ast::Value> value =
         node->body().consume_unchecked<ValueBody>();
-    StatementsBody new_body;
+    std::vector<std::unique_ptr<ast::Statement>> new_body;
     new_body.emplace_back(std::make_unique<ast::ReturnStatement>(
         value->location(), std::move(value)));
-    node->body() = std::move(new_body);
+    node->body() = std::make_unique<ast::Block>(value->location(), std::move(new_body));
```

Comment:
can't we give back directly a return statement ? Would the code gen handle it correctly ?